plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -380,7 +379,7 @@ public MachineLearningPlugin(Settings settings) {
 
     @Override
     public List<ActionHandler<? extends ActionRequest, ? extends ActionResponse>> getActions() {
-        return ImmutableList
+        return List
```

Comment:
According to PE we should avoid using class like `ImmutableList` `ImmutableMap`, instead if we want to create an immutable type we should use java.utils directly

Reply:
`ImmutableList` is inherently thread-safe. Multiple threads can access it without requiring synchronization. A standard List is not thread-safe by default. Did you check if it can arise any issue? Let's not change this in this PR if we are unsure about the further consequences. If we need to change this, we can raise a separate PR.

Reply:
Ok will revert